appserver/tests/ctsrq.sh

```diff
@@ -0,0 +1,206 @@
+#!/bin/bash -e
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+#
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved.
+#
+# The contents of this file are subject to the terms of either the GNU
+# General Public License Version 2 only ("GPL") or the Common Development
+# and Distribution License("CDDL") (collectively, the "License").  You
+# may not use this file except in compliance with the License.  You can
+# obtain a copy of the License at
+# https://glassfish.dev.java.net/public/CDDL+GPL_1_1.html
+# or packager/legal/LICENSE.txt.  See the License for the specific
+# language governing permissions and limitations under the License.
+#
+# When distributing the software, include this License Header Notice in each
+# file and include the License file at packager/legal/LICENSE.txt.
+#
+# GPL Classpath Exception:
+# Oracle designates this particular file as subject to the "Classpath"
+# exception as provided by Oracle in the GPL Version 2 section of the License
+# file that accompanied this code.
+#
+# Modifications:
+# If applicable, add the following below the License Header, with the fields
+# enclosed by brackets [] replaced by your own identifying information:
+# "Portions Copyright [year] [name of copyright owner]"
+#
+# Contributor(s):
+# If you wish your version of this file to be governed by only the CDDL or
+# only the GPL Version 2, indicate your decision by adding "[Contributor]
+# elects to include this software in this distribution under the [CDDL or GPL
+# Version 2] license."  If you don't indicate a single choice of license, a
+# recipient has the option to distribute your version of this file under
+# either the CDDL, the GPL Version 2 or to extend the choice of license to
+# its licensees as provided above.  However, if you add GPL Version 2 code
+# and therefore, elected the GPL Version 2 license, then the option applies
+# only if the new code is made subject to such option by the copyright
+# holder.
+
+################################## Constants ##################################
+
+declare -r USAGE=$(cat <<'EOF'
+Prerequites: User must specify CTS_REMOTE_QUEUE_URL before running this script. This URL must point to the upstream hudson job of CTS Remote Queue.
+
+Usage:
+ 1. ctsrq.sh -l ---> List all the available CTS test identifiers (both top-level and inner-level) without running them
+ 2. ctsrq.sh -s ---> List all the available top-level CTS test identifiers without running them
+ 3. ctsrq.sh -i ---> List the inner-level CTS test identifiers for a given top-level CTS test identifier without running them
+ 4. ctsrq.sh -b <Glassfish_Branch> -a [-e "<email-id1>, <email-id2>, .."] ---> Run all the available top-level CTS test identifiers against the specified Glassfish branch and email the results to the specified email-ids
```

Comment:
Instead of mentioning the email id in CLI, IMO it would be a good idea to define the email id in .bashrc and read it from there.